packages/hadron-express/src/hadronToExpress.ts

```diff
@@ -87,22 +91,118 @@ const convertToExpress = (
     promises.push(jsonProvider(paths, extensions));
   }
 
-  return Promise.all(promises)
-    .then((results) =>
-      results.reduce(
-        (aggregation, current) => ({ ...aggregation, ...current }),
-        {},
-      ),
-    )
-    .then((routes: any) => {
-      (Object as any).keys(routes).map((key: string) => {
-        const route: IRoute = routes[key];
-        validateMethods(key, route.methods);
-        const middlewares: Middleware[] =
-          generateMiddlewares(route, containerProxy) || [];
-        createRoutes(app, route, middlewares, containerProxy, key);
-      });
-    });
+  return (
+    Promise.all(promises)
+      .then((results) =>
+        results.reduce(
+          (aggregation, current) => ({ ...aggregation, ...current }),
+          {},
+        ),
+      )
+      // flatten routes and prepare all components
+      .then((routes: any) =>
+        (Object as any)
+          .keys(routes)
+          .map((key: string) => prepareRoute(routes[key], key, containerProxy))
+          .reduce((agg: any, current: any) => ({ ...agg, ...current })),
+      )
+      .then((preparedRoutes: IRoutesConfig) => {
+        (Object as any).keys(preparedRoutes).map((key: string) => {
+          const route: IRoute = preparedRoutes[key];
+          createRoutes(app, route, container, key);
+        });
+      })
+  );
+};
+
+const prepareRoute = (
+  route: IRoute,
+  key: string,
+  container: IContainer,
+  parentRoute: IRoute = {},
+  parentKey: string = null,
+) => {
+  const middlewares = prepareMiddlewares(
+    generateMiddlewares(route.middleware, container),
+    generateMiddlewares(route.$middleware, container),
+    parentRoute.middleware,
+  );
+  const path = preparePath(route.path, route.$path, parentRoute.path);
+
+  const methods = prepareMethods(
+    route.methods,
+    route.$methods,
+    parentRoute.methods,
+  );
+
+  const routeKey = parentKey ? `${parentKey}.${key}` : key;
+
+  const preparedRoute = {
+    ...route,
+    path,
+    methods,
+    middleware: middlewares,
+  };
+
+  const result = {
+    [routeKey]: preparedRoute,
+  };
+
+  if (route.routes) {
+    return {
+      ...result,
+      ...(Object as any)
+        .keys(route.routes)
+        .map((childKey: string) =>
+          prepareRoute(
+            (route.routes as any)[childKey],
+            childKey,
+            container,
+            preparedRoute,
+            routeKey,
+          ),
+        )
+        .reduce((agg: any, current: any) => ({ ...agg, ...current })),
```

Comment:
I'm for using whole words as variable names with only exceptions being very widely used ones like `req`, `res`

IMO `agg` should be named `accumulator` (or `aggregator` if you want)

Reply:
It's more like esthetic change, but you are right.